include/eosio/vm/opcodes_def.hpp

```diff
@@ -1,290 +1,316 @@
+   opcode_macro(empty0xDA, 0xDA, "")   \
+   opcode_macro(empty0xDB, 0xDB, "")   \
+   opcode_macro(empty0xDC, 0xDC, "")   \
+   opcode_macro(empty0xDD, 0xDD, "")   \
+   opcode_macro(empty0xDE, 0xDE, "")   \
+   opcode_macro(empty0xDF, 0xDF, "")   \
+   opcode_macro(empty0xE0, 0xE0, "")   \
+   opcode_macro(empty0xE1, 0xE1, "")   \
+   opcode_macro(empty0xE2, 0xE2, "")   \
+   opcode_macro(empty0xE3, 0xE3, "")   \
+   opcode_macro(empty0xE4, 0xE4, "")   \
+   opcode_macro(empty0xE5, 0xE5, "")   \
+   opcode_macro(empty0xE6, 0xE6, "")   \
+   opcode_macro(empty0xE7, 0xE7, "")   \
+   opcode_macro(empty0xE8, 0xE8, "")   \
+   opcode_macro(empty0xE9, 0xE9, "")   \
+   opcode_macro(empty0xEA, 0xEA, "")   \
+   opcode_macro(empty0xEB, 0xEB, "")   \
+   opcode_macro(empty0xEC, 0xEC, "")   \
+   opcode_macro(empty0xED, 0xED, "")   \
+   opcode_macro(empty0xEE, 0xEE, "")   \
+   opcode_macro(empty0xEF, 0xEF, "")   \
+   opcode_macro(empty0xF0, 0xF0, "")   \
+   opcode_macro(empty0xF1, 0xF1, "")   \
+   opcode_macro(empty0xF2, 0xF2, "")   \
+   opcode_macro(empty0xF3, 0xF3, "")   \
+   opcode_macro(empty0xF4, 0xF4, "")   \
+   opcode_macro(empty0xF5, 0xF5, "")   \
+   opcode_macro(empty0xF6, 0xF6, "")   \
+   opcode_macro(empty0xF7, 0xF7, "")   \
+   opcode_macro(empty0xF8, 0xF8, "")   \
+   opcode_macro(empty0xF9, 0xF9, "")   \
+   opcode_macro(empty0xFA, 0xFA, "")   \
+   opcode_macro(empty0xFB, 0xFB, "")   \
+   opcode_macro(empty0xFC, 0xFC, "")   \
+   opcode_macro(empty0xFD, 0xFD, "")   \
+   opcode_macro(empty0xFE, 0xFE, "")
+#define EOS_VM_ERROR_OPS(opcode_macro) \
+   opcode_macro(error, 0xFF, "error")
 
 /* clang-format on */
 
-#define EOS_VM_CREATE_ENUM(name, code) name = code,
+namespace eosio { namespace vm { namespace detail {
+   template <std::size_t I, std::size_t Name_N, std::size_t Fmt_N>
+   inline const char* format_str_impl(const char (&name)[Name_N], const char (&fmt)[Fmt_N]) {
+      return fmt;
+   }
+   template <std::size_t I, std::size_t Name_N, std::size_t Fmt_N, typename Arg, typename... Args>
+   inline const char* format_str_impl(const char (&name)[Name_N], const char (&fmt)[Fmt_N], Arg&& arg, Args&&... args) {
+      if constexpr (I == Fmt_N) {
+         return "";
+      } else {
+         if (fmt[I] == '%')
+            return std::string(std::to_string(std::forward<Arg>(arg)) + format_str_impl<I+1>(name, fmt, std::forward<Args>(args)...)).c_str();
+         else if (fmt[I] == '!')
+            return std::string(std::string(name) + format_str_impl<I+1>(name, fmt, std::forward<Args>(args)...)).c_str();
+         else
+            return std::string(std::string(1, fmt[I]) + format_str_impl<I+1>(name, fmt, std::forward<Args>(args)...)).c_str();
+      }
+   }
 
-#define EOS_VM_CREATE_STRINGS(name, code) #name,
+   template <std::size_t Name_N, std::size_t Fmt_N, typename... Args>
+   inline const char* fmt_str(const char (&name)[Name_N], const char (&fmt)[Fmt_N], Args&&... args) {
```

Comment:
Or alternatively, you could just remove that one line from `debug_visitor.hpp` (since this whole thing seems incomplete) and revert all these `fmt` additions in this PR. None of that is relevant for the core changes of this PR, so it can be properly handled later.